superset/db_engine_specs/base.py

```diff
@@ -190,10 +205,27 @@ def get_dbapi_exception_mapping(cls) -> Dict[Type[Exception], Type[Exception]]:
 
         :return: A map of driver specific exception to superset custom exceptions
         """
-        return {}
+        dbapi_exception_names = {
+            "Error": SupersetDBAPIError,
+            "InterfaceError": SupersetDBAPIInterfaceError,
+            "DatabaseError": SupersetDBAPIDatabaseError,
+            "DataError": SupersetDBAPIDataError,
+            "OperationalError": SupersetDBAPIOperationalError,
+            "IntegrityError": SupersetDBAPIIntegrityError,
+            "InternalError": SupersetDBAPIInternalError,
+            "ProgrammingError": SupersetDBAPIProgrammingError,
+            "NotSupportedError": SupersetDBAPINotSupportedError,
+        }
+        return {
+            getattr(dbapi, exception_name): exception
+            for exception_name, exception in dbapi_exception_names.items()
+            if hasattr(dbapi, exception_name)
+        }
```

Comment:
I wonder how common it is for dbapi's to expose the exceptions in the main module? I quickly checked `pydruid` and `pyhive`, and they seem to export them in `pydruid.db.exceptions` and `pyhive.exc` respectively. We could potentially provide an optional path to the exception module of the specific spec.

Reply:
I think in theory they should be exposed in the main module:

> The module should make all error information available through these exceptions or subclasses thereof ([ref](https://www.python.org/dev/peps/pep-0249/#exceptions))

From the context it seems to me that "module" means the top-level module, but if implementations are not following that then it really doesn't matter. :-/

It looks like SQLAlchemy has something similar, let me see how they do it:

https://github.com/sqlalchemy/sqlalchemy/blob/fcf539d090a95fb179ca03beffd10122e97aa002/lib/sqlalchemy/engine/default.py#L203-L206

Reply:
Ha, it looks very similar: https://github.com/sqlalchemy/sqlalchemy/blob/640cd8a70f8a664b7834c5f74ec322fdea644043/lib/sqlalchemy/dialects/postgresql/pypostgresql.py#L90-L110

But it's defined per dialect. I'll try to leverage that.

Reply:
For context, SQLAlchemy has a mapping between the exception **name** and a SQLAlchemy exception, which makes it easier: https://github.com/sqlalchemy/sqlalchemy/blob/1e75c189da721395bc8c2d899c722a5b9a170404/lib/sqlalchemy/exc.py#L542